app/controllers/media_controller.rb

```diff
@@ -47,7 +58,31 @@ def stacks_media_stream_params
     allowed_params.slice(:id, :file_name, :format)
   end
 
+  def id
+    allowed_params[:id]
+  end
+
+  def file_name
+    allowed_params[:file_name]
+  end
+
   def load_media
     @media ||= StacksMediaStream.new(stacks_media_stream_params)
   end
+
+  def token_still_valid?(token_string, expected_id, expected_file_name, expected_user_ip_addr)
```

Comment:
sure, i'd do that if your preference for the change was strong.  i like the `still` since it implies that the token won't be valid forever, and was actually debating renaming the `verify_encrypted_token_string` method along those lines, but couldn't think of a better name for that one.